mimic/model/heat_objects.py

```diff
@@ -9,6 +9,27 @@
 from mimic.model.behaviors import BehaviorRegistryCollection, EventDescription
 
 
+def get_stack_or_404(f):
+    """
+    Decorator which retrieves a stack from a collection and adds it to the
+    function's arguments if it exists. If the stack does not exist, a 404
```

Comment:
If you're going to leave it as a decorator - not a dealbreaker for me, just not my preference - make sure the docstring shows *both* how the function is declared *and* how it is called, and explain the distinction.